CMakeLists.txt

```
@@ -36,6 +47,13 @@ if(TBB_FOUND)
     "[${TBB_LIBRARY}][${TBB_INCLUDE_DIR}]")
 endif()
 
+# go find nanobind
+execute_process(
+  COMMAND ${Python_EXECUTABLE} "-m" "nanobind" "--cmake_dir"
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+  OUTPUT_VARIABLE nanobind_ROOT)
+find_package(nanobind CONFIG REQUIRED)
+
```

```suggestion
if (nanobind_FOUND)
    find_package_message(nanobind "Found nanobind: ${nanobind_DIR} ${nanobind_VERSION}" "[${nanobind_DIR},${nanobind_VERSION}]")
endif()
```
I forgot to add this earlier. It prints a status message indicating which nanobind was found. This is immensely useful for debugging CMake builds.

Once https://github.com/wjakob/nanobind/pull/618 (or an alternative solution) is merged and a new nanobind release is pushed, then we can also add the version requirement to `find_package`:
```
find_package(nanobind 2.0 CONFIG REQUIRED)
```